vtex/faststore
# feat(collections): Source collections from category tree

## What's the purpose of this pull request?
This PR is the first iteration after this discussion #825 . 
This PR makes `StoreCollection` to be sourced by `@vtex/gatsby-source-vtex` for Brands/Categories. Also, this exports the `PLP` contentType from the CMS and adds first class support to the catalog in the CMS plugin. 

## How it works
`@vtex/gatsby-source-vtex` has gained a new parameter. This parameter is called: `sourceCollections`. When this param is set, the plugin will list all categories/brands and create a corresponding `StoreCollection` for each. 

`@vtex/gatsby-plugin-cms` now exports a new contentType called `PLP`. When sourcing data, if `gatsby-plugin-cms` encounters an instance `PLP`, it will get the corresponding `StoreCollection` and add this `PLP` as a foreign key of `StoreCollection` and add the corresponding SearchParams.

In the end, the store will have the following schema:
![image](https://user-images.githubusercontent.com/1753396/127517266-5df5c9cb-8611-4fad-be94-62f168709653.png)

## Future Work
As you can see, this is only a partial implementation of what's described on #825 and requires more PRs to fully implement the RFC. I intend to do more PRs later to finish implementing the other use cases discussed on #825 and incremental builds. I didn't do all in one single PR because this would create a HUGE PR. 

## How to test it
The sole intent of this PR is to create new data to the Gatsby's data layer and nothing should change in any store having this version. Make sure this is the case for:
https://github.com/vtex-sites/storecomponents.store/pull/1060
https://github.com/vtex-sites/btglobal.store/pull/729
https://github.com/vtex-sites/marinbrasil.store/pull/572
